src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/validation/HaFlowValidationService.java

```diff
@@ -0,0 +1,224 @@
+import org.openkilda.wfm.error.FlowNotFoundException;
+import org.openkilda.wfm.error.IllegalFlowStateException;
+import org.openkilda.wfm.error.SwitchNotFoundException;
+import org.openkilda.wfm.topology.flowhs.fsm.validation.SimpleSwitchRule;
+import org.openkilda.wfm.topology.flowhs.fsm.validation.SimpleSwitchRuleComparator;
+import org.openkilda.wfm.topology.flowhs.fsm.validation.SimpleSwitchRuleConverter;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.nio.file.InvalidPathException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class HaFlowValidationService {
+
+    private final HaFlowRepository haFlowRepository;
+    private final SimpleSwitchRuleConverter simpleSwitchRuleConverter = new SimpleSwitchRuleConverter();
+    private final SimpleSwitchRuleComparator simpleSwitchRuleComparator;
+    private final RuleManager ruleManager;
+    private final PersistenceManager persistenceManager;
+
+    public HaFlowValidationService(@NonNull PersistenceManager persistenceManager,
+                                   RuleManager ruleManager) {
+        this.haFlowRepository = persistenceManager.getRepositoryFactory().createHaFlowRepository();
+        this.ruleManager = ruleManager;
+        this.persistenceManager = persistenceManager;
+        SwitchRepository switchRepository = persistenceManager.getRepositoryFactory().createSwitchRepository();
+        this.simpleSwitchRuleComparator = new SimpleSwitchRuleComparator(switchRepository);
+    }
+
+    /**
+     * Check the flow status.
+     */
+    public HaFlow checkFlowStatusAndGetFlow(String flowId) throws FlowNotFoundException, IllegalFlowStateException {
+        HaFlow flow = haFlowRepository.findById(flowId).orElseThrow(() -> new FlowNotFoundException(flowId));
+        if (FlowStatus.DOWN.equals(flow.getStatus())) {
+            throw new IllegalFlowStateException(flowId);
+        }
+        return flow;
+    }
+
+    /**
+     * Validate flow.
+     */
+    public HaFlowValidationResponse validateFlow(String flowId, List<FlowDumpResponse> flowDumpResponses,
+                                                 List<MeterDumpResponse> metersDumpResponses,
+                                                 List<GroupDumpResponse> groupDumpResponses,
+                                                 Set<SwitchId> switchIdSet)
+            throws FlowNotFoundException, SwitchNotFoundException {
+
+        Optional<HaFlow> foundFlow = haFlowRepository.findById(flowId);
+        if (!foundFlow.isPresent()) {
+            throw new FlowNotFoundException(flowId);
+        }
+        HaFlow flow = foundFlow.get();
+
+        if (flow.getForwardPath() == null) {
+            throw new InvalidPathException(flowId, "Forward path was not returned.");
```

Comment:
Thank you, 
Regarding this:
"I'd suggest not to throw exceptions here, but collect all errors and put them to the response (maybe to a log if they are not important for the user)."

It looks like a significant change, I need to discuss it with Sergey.

'path' fixed:)

Reply:
We have to keep this behaviour since it cause the changes for API.
More over, is the getForwardPath is null it should be the internal server error.. 
 it is not a user problem that we can not find the path in DB, it is 500 problem..